source/base/OwnedBuffer.ooc

```diff
@@ -0,0 +1,106 @@
+/*
+* Copyright (C) 2015 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use ooc-base
+import math
+
+OwnedBuffer: cover {
+	_pointer: UInt8*
+	_size: Int
+	_owner: Owner
+	pointer ::= this _pointer
+	size ::= this _size
+	owner ::= this _owner
+	isOwned ::= this _owner != Owner Unknown && this _owner != Owner Static && this _owner != Owner Stack && this _pointer != null
```

Comment:
Would it make sense to put the part of this logic that compares `this _owner` to various `Owner` variants in its own function?